src/main/java/com/github/onsdigital/babbage/util/http/BabbageHttpClient.java

```diff
@@ -0,0 +1,130 @@
+package com.github.onsdigital.babbage.util.http;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.conn.HttpClientConnectionManager;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.impl.conn.PoolingHttpClientConnectionManager;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Class for building Closable Http clients to be used by Babbage
+ */
+public class BabbageHttpClient implements AutoCloseable {
+
+    protected final CloseableHttpClient httpClient;
+    protected final URI HOST;
+    private final PoolingHttpClientConnectionManager connectionManager;
+    private final IdleConnectionMonitorThread monitorThread;
+
+    public BabbageHttpClient(String host, ClientConfiguration configuration) {
+        HOST = resolveHostUri(host);
```

Comment:
Be consistent use `this` on the left hand of all assignments to the object in the constructor.